src/deck/guides/3.0-upgrade.md

```diff
@@ -0,0 +1,85 @@
+This will break all regex routing in 3.x.
+
+## decK command behavior with 3.x
+
+When running decK commands against {{site.base_gateway}} 3.x, take note of the following behavior:
+
+`deck dump`
+: Explicitly sets the format version to 3.0.
+It assumes that the paths have been correctly transformed, either via Kong migrations, manually, or through a `deck convert`.
+
+`deck diff`, `deck validate` (with `--online` flag only), or `deck sync`
+: decK performs a check to ensure all regex routes are correctly prefixed.
+If not, the behavior depends on the control plane type:
+* Self-managed {{site.base_gateway}} 2.x (`_format_version: 1.1` or earlier): Prints an error and stops the operation.
+* Self-managed {{site.base_gateway}} 3.x (`_format_version: 3.0`): Prints a warning and goes ahead with the `diff`, `sync`, or `validate` operation as usual.
+* {{site.konnect_short_name}}: Prints a warning and goes ahead with the `diff`, `sync`, or `validate` operation as usual.
+
+`deck convert`
+: Includes `--from` and `--to` flags for converting state files between major versions.
+Converts all relevant files in the present working directory:
+  * Upgrades the `_format_version` setting to `3.0`
+  * Adds the `~` prefix to all route paths containing a regex pattern
+
+: You can optionally provide `--input-file` and `--output-file` flags to limit conversion to
+a subset of files.
+
+### Using decK with {{site.konnect_short_name}} data planes
+
+{{site.konnect_short_name}} supports 3.x and 2.x data planes, but the {{site.konnect_short_name}} control plane version is 3.x.
+Since decK can't tell if a runtime group is intended for 2.x or 3.x data planes, it will always dump configuration with `_format_version: 3.0`.
+
+To avoid compatibility errors, make sure that all data planes in a single runtime group are of the same major version (all 2.x or all 3.x).
+
+For all `diff`, `sync`, or `validate` (with `--online` flag) operations against {{site.konnect_short_name}}, decK issues warnings when it detects incorrect regex path configurations.
+
+## Convert 2.x state file to 3.x
+
+1. Run `deck-convert` against your 2.x state file to turn it into a 3.x file:
+
+    ```sh
+    deck convert \
+    --from kong-gateway-2.x \
+    --to kong-gateway-3.x \
+    --input-file kong.yaml \
+    --output-file new-kong.yaml
+    ```
+
+    You can leave `input-file` out to convert all declarative configuration files in your working directory.
+    We recommend specifying both input and output files.
+
+
+2. Validate the converted files in a test environment.
+
+    These changes may not be fully correct or exhaustive, so perform manual audits of the generated file before applying
+    the configuration in production.
+
+## FAQ
+
+**What happens if I set the format version to 3.0 but don't update regex paths?**
+
+The state file will be broken. Avoid setting the format version manually.
```

Comment:
```suggestion
When you sync the file, decK will issue a warning stating that an invalid regex pattern was detected and that it must be updated with a `~` to distinguish it from a standard prefix match. 
```

My understanding is that if you do this, it will still work. The state is broken, but this will overwrite your database values and you will not be able to access regex routes anymore.